includes/admin/class-wc-payments-bnpl-announcement.php

```diff
@@ -0,0 +1,241 @@
+<?php
+/**
+ * Class WC_Payments_Bnpl_Announcement
+ *
+ * @package WooCommerce\Payments\Admin
+ */
+
+/**
+ * WC_Payments_Bnpl_Announcement class.
+ */
+class WC_Payments_Bnpl_Announcement {
+
+	/**
+	 * WC_Payment_Gateway_WCPay.
+	 *
+	 * @var WC_Payment_Gateway_WCPay
+	 */
+	private $gateway;
+
+	/**
+	 * WC_Payments_Account.
+	 *
+	 * @var WC_Payments_Account
+	 */
+	private $account;
+
+	/**
+	 * The current time. Useful for mocking in tests.
+	 *
+	 * @var int
+	 */
+	private $current_time;
+
+	/**
+	 * Initialize class actions.
+	 *
+	 * @param WC_Payment_Gateway_WCPay $gateway Payment Gateway.
+	 * @param WC_Payments_Account      $account Account information.
+	 * @param int                      $current_time The current time.
+	 */
+	public function __construct( WC_Payment_Gateway_WCPay $gateway, WC_Payments_Account $account, int $current_time ) {
+		$this->gateway      = $gateway;
+		$this->account      = $account;
+		$this->current_time = $current_time;
+	}
+
+	/**
+	 * Initializes this class's WP hooks.
+	 *
+	 * @return void
+	 */
+	public function init_hooks() {
+		add_action( 'current_screen', [ $this, 'maybe_enqueue_scripts' ] );
+	}
+
+	/**
+	 * Needs to run after `current_screen`, to determine which page we're currently on.
+	 *
+	 * @return void
+	 */
+	public function maybe_enqueue_scripts() {
+		if ( ! is_admin() ) {
+			return;
+		}
+
+		// Only shown once to each Administrator and Shop Manager users.
+		if ( ! current_user_can( 'manage_woocommerce' ) ) {
+			return;
+		}
+
+		// Time boxed - Campaign expires after 90 days.
+		if ( $this->current_time > strtotime( '2024-07-15 00:00:00' ) ) {
+			return;
+		}
+
+		// Only displayed to BNPL eligible countries - AU, NZ, US, AT, BE, CA, CZ, DK, FI, FR, DE, GR, IE, IT, NO, PL, PT, ES, SE, CH, NL, UK, US.
+		if ( ! in_array(
+			$this->account->get_account_country(),
```

Comment:
This check uses the cached account data, so we should be ok in terms of performance.